Mark legacy directories as linguist-vendored

Monkatraz opened the pull request:
I think it's a little unfair to include the ginormous pile of legacy Wikidot code in our project's linguist data. Vendored files will still get included in diffs, it's just that we won't see this anymore:
![image](https://user-images.githubusercontent.com/34875062/138369416-b3382ca4-5c6b-4bd1-8d9e-934665bcdad4.png)

Smarty is 6.8%. That's painful.

Monkatraz committed:
mark legacy directories as linguist-vendored

emmiegit reviewed (changes requested):
I disagree with this change, the language bar should get cut as we go and actually delete legacy files.

In this way it's actually valuable for seeing where our repo is at in terms of remaining legacy nonsense to cut, and a sense of accomplishment when we can submit a PR to delete old garbage. The smarty stuff for instance will disappear when we can make the final switchover to the new frontend, and the % of PHP will continue to shrink as we delete legacy code and rewrite stuff in other services.

Monkatraz commented:
Interesting, considering `web/**` was already ignored. But alright.

emmiegit commented:
Web was being ignored? I wasn't aware of that.

Monkatraz commented:
Yeah, it is. Well, actually, it was originally `*.js` and `*.css`, but when we moved `client/` into `web/`, I changed it to `web/**` (as that was the intention of ignoring JS and CSS files).
